src/function.js

```diff
@@ -1,10 +1,35 @@
 /** @license MIT License (c) copyright 2010-2016 original author or authors */
 
 // id :: a -> a
-export const id = x => x;
+export const id = x => x
 
 // compose :: (b -> c) -> (a -> b) -> (a -> c)
-export const compose = (f, g) => x => f(g(x));
+export const compose = (f, g) => x => f(g(x))
 
 // apply :: (a -> b) -> a -> b
-export const apply = (f, x) => f(x);
+export const apply = (f, x) => f(x)
+
+// curry2 :: (a -> b -> c) -> a -> b -> c
```

Comment:
I don't know if it's even possible to write an accurate type definition in Haskell syntax for JS curry :)  In Haskell, `uncurry` is `(a -> b -> c) -> ((a, b) -> c)`.  The critical bit of that is that it "transforms" one `->` into `,`.  IOW, it turns a function call into a pair.

Curry is the inverse of uncurry, and flipping uncurry around yields: `curry :: ((a, b) -> c) -> (a -> b -> c)`.

So, maybe that?

```
curry2 :: ((a, b) -> c) -> (a -> b -> c)
curry3 :: ((a, b, c) -> d) -> (a -> b -> c -> d)
```

Thoughts?